src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/GetKeyworkerAllocations.kt

```diff
@@ -5,21 +5,60 @@ import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerAllocationReposito
 import uk.gov.justice.digital.hmpps.keyworker.domain.ReferenceData
 import uk.gov.justice.digital.hmpps.keyworker.dto.Actioned
 import uk.gov.justice.digital.hmpps.keyworker.dto.CodedDescription
+import uk.gov.justice.digital.hmpps.keyworker.dto.CurrentAllocation
+import uk.gov.justice.digital.hmpps.keyworker.dto.CurrentKeyworker
+import uk.gov.justice.digital.hmpps.keyworker.dto.CurrentPersonStaffAllocation
 import uk.gov.justice.digital.hmpps.keyworker.dto.Keyworker
 import uk.gov.justice.digital.hmpps.keyworker.dto.KeyworkerAllocation
 import uk.gov.justice.digital.hmpps.keyworker.dto.PersonStaffAllocationHistory
+import uk.gov.justice.digital.hmpps.keyworker.events.ComplexityOfNeedLevel
 import uk.gov.justice.digital.hmpps.keyworker.integration.ManageUsersClient
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.CaseNotesApiClient
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.UsageByPersonIdentifierRequest.Companion.sessionTypes
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.summary
 import uk.gov.justice.digital.hmpps.keyworker.sar.StaffSummary
 import uk.gov.justice.digital.hmpps.keyworker.sar.internal.StaffDetailProvider
+import java.time.LocalDate.now
 
 @Service
 class GetKeyworkerAllocations(
+  private val complexityOfNeed: ComplexityOfNeedGateway,
   private val allocationRepository: KeyworkerAllocationRepository,
   private val manageUsers: ManageUsersClient,
   private val staffDetailProvider: StaffDetailProvider,
   private val prisonService: PrisonService,
+  private val caseNotesApiClient: CaseNotesApiClient,
 ) {
-  fun allocationHistoryFor(prisonNumber: String): PersonStaffAllocationHistory {
+  fun currentFor(prisonNumber: String): CurrentPersonStaffAllocation {
+    val level = complexityOfNeed.getOffendersWithMeasuredComplexityOfNeed(setOf(prisonNumber)).firstOrNull()?.level
```

Comment:
Since the intended usage of this endpoint is the prisoner profile, it makes sense to reduce the load on complexity of need as much as possible. It should only make this call when the prison is set to have HCN prisoners I reckon. Could be done in the complexity of need service perhaps?